src/transformers/quantizers/quantizer_hqq.py

```diff
@@ -0,0 +1,177 @@
+        If not, we create a temp linear layer with the module state_dict params and use it for quantization
+        """
+
+        module, tensor_name = get_module_from_name(model, param_name)
+
+        if type(module) is not torch.nn.Linear:
+            return
+
+        layer_name = param_name.replace(".weight", "").replace(".bias", "")
+        parent_module = find_parent(model, layer_name)
+        node = layer_name.split(".")[-1]
+
+        # Step 0: set module state_dict
+        module_state_dict = {key.split(".")[-1]: state_dict[key] for key in state_dict if layer_name in key}
+
+        # Step 1: Check if the state_dict of the module already contains quantized parameters
+        if ("W_q" in module_state_dict) and ("meta" in module_state_dict):
+            module_hqq = HQQLinear(
+                linear_layer=None, quant_config=None, compute_dtype=self.torch_dtype, device=target_device
+            )
+            module_hqq.load_state_dict(module_state_dict)
+            setattr(parent_module, node, module_hqq)
+            return
+
+        # Step 2: populate module with weight/bias from module state dict
+        for key in module_state_dict:
+            setattr(module, key, torch.nn.Parameter(module_state_dict[key]))
+
+        """
+        Step 3: Replace module with either HQQLinear or move it to device. We do this via setattr on the parent as doing on it on the module
+        directly doesn't work.
+        """
+
+        if hasattr(module, "quant_config"):
+            setattr(
+                parent_module,
+                node,
+                HQQLinear(
+                    module,
+                    module.quant_config,
+                    compute_dtype=self.torch_dtype,
+                    device=target_device,
+                    del_orig=True,
+                ),
+            )
+        else:
+            setattr(parent_module, node, module.to(self.torch_dtype).to(target_device))
+
+        torch.cuda.empty_cache()
+
+    def update_torch_dtype(self, torch_dtype: "torch.dtype") -> "torch.dtype":
+        if torch_dtype is None:
+            torch_dtype = torch.float16
+        return torch_dtype
+
+    def _process_model_before_weight_loading(
+        self,
+        model: "PreTrainedModel",
+        device_map,
+        keep_in_fp32_modules: List[str] = [],
+        **kwargs,
+    ):
+        # Add the corresponding quant_config to each valid module. This allows us to do the actual nn.Linear > HQQLinear in create_quantized_param()
+        model = prepare_for_hqq_linear(model, quantization_config=self.quantization_config)
+
+        # model.config.quantization_config is done inside prepare_for_hqq_linear
+
+    def _process_model_after_weight_loading(self, model: "PreTrainedModel", **kwargs):
+        model.is_hqq_quantized = True
+        model.is_hqq_serializable = self.is_serializable
+        return model
+
+    @property
+    def is_serializable(self):
+        return True
```

Comment:
```suggestion
        return False
```
See my suggestion above, I think we should have a v1 without serialization first